homeassistant/components/samsungtv/bridge.py

```diff
@@ -359,6 +353,22 @@ def _get_remote(self, avoid_open: bool = False) -> Remote:
                 self._remote = None
         return self._remote
 
+    def is_on(self) -> bool:
+        """Tells if the TV is on."""
+        if self._remote is not None:
+            self.close_remote()
+        try:
+            info = self.device_info()
+            if info is not None and isinstance(info["device"]["PowerState"], str):
```

Comment:
We need a test for this and one for when the key isn’t present as not all models are likely to have this given the wide array of firmwares

Reply:
Not really sure what tests should I add. Should I add tests for the state with and without `PowerState`? I tried to take a look at the existing tests but I'm a bit confused.

I enabled edits in case someone wants to point me in the right direction 😄

Reply:
You already have tests without `PowerState`, so you shouldn't change those (ensures compatibility).
Instead you should add a single new test with PowerState added to the device info.